assets/dev/js/editor/controls/media.js

```diff
@@ -243,6 +251,54 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
 	onBeforeDestroy() {
 		this.$el.remove();
 	},
+
+	async onMediaImageSize() {
+		this.ui.mediaImageSize.css( 'color', 'inherit' );
+		const previewPlaceHolder = this.$el.find( '.elementor-control-media__preview' );
+		const selectedImageId = {
+			id: this?.getControlValue( 'id' ) || previewPlaceHolder[ 0 ].getAttribute( 'data-id' ),
+			alt: this?.getControlValue( 'alt' ) || previewPlaceHolder[ 0 ].getAttribute( 'alt' ),
+		};
+
+		const selectedSize = this.ui.mediaImageSize.val();
+		let imageURL = await elementor.imagesManager.getImageUrl( {
+			id: selectedImageId.id,
+			size: selectedSize,
+		} );
+		const stateOptions = {
+			url: null,
+			id: selectedImageId.id,
+			alt: selectedImageId.alt,
+			dimensions: selectedSize,
+			source: 'library',
+		};
+		if ( imageURL ) {
+			stateOptions.url = imageURL;
+			this.setValue( stateOptions );
+		} else {
+			elementor.channels.editor.once( 'imagesManager:detailsReceived', ( data ) => {
```

Comment:
You might have a race condition here, where you miss this event since it already finished before adding this listener. 
Maybe always add this event before calling `getImageUrl()`?
Or maybe always throw the same event when the value is returned from cache, and you'll be able to use the same mechanism for both?

Something like this:
```JS
elementor.channels.editor.once( 'imagesManager:detailsReceived', ( data ) => {
    this.setValue( { /* ... */ } );
} );

elementor.imagesManager.getImageUrl( ... );
```

Anyway, looks like the images manager will throw this event only if the panel is open (and when using "custom" size):
https://github.com/elementor/elementor/blob/39ffa61c3da41fc4ae87e97ba0a33ee3d6f51b8d/assets/dev/js/editor/utils/images-manager.js#L58-L65

So you can have another bug there. worth checking

Reply:
The first run triggers only the hook.
Then the function return the value and without the hook. 
The hook runs only on the first call when `getImageUrl` is actually request the image data.

Reply:
1. It is not matter of fast request, The imagesManager has a cache mechanism that will always trigger the `imagesManager:detailsReceived` hook on the first request. And later on from cache object.
As you can see here :

https://github.com/elementor/elementor/blob/945ede74b8cfcff4478f8f5d0d752534e18930fa/assets/dev/js/editor/utils/images-manager.js#L150

 I also consulted with Mati and Nevo about this matter, this is the solution we found in the configuration that the imagesManager API is currently working on.

2. the images size currently not does not support custom size, So I guess we'll have to deal with the moment once we implement support for custom size.